app/clients/handong_feed_app_client.py

```
@@ -185,4 +185,41 @@ def get_latest_for_date(self) -> SubjectTagDto.GetLatestForDateResDto:
+            # 예상한 204 No Content 응답 처리
+            if response.status_code == 204:
+                return
+
+            # 예상 외의 정상 응답 (ex: 200, 201)이 오면 의심
+            raise Exception(f"Unexpected success response: {response.status_code}")
+
+        except requests.exceptions.HTTPError as e:
+            status_code = e.response.status_code
+
+            if status_code == 404:
+                raise Exception(f"Subject not found with ID: {subject_id}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)